### PR TITLE
[django] Detect wsgi application and set it on Dockerfile

### PR DESCRIPTION
**Pull request by katiayn**
Find `<project_name>/wsgi.py` file and set `project_name` on Dockefile:

```
CMD ["gunicorn", "--bind", ":8000", "--workers", "2", "{{ .wsgiName }}.wsgi"]
```

### CODE REVIEW

**Review by dangra — COMMENTED** [review]
love the simplicity of this one, left a few comments.

File: scanner/django.go [hunk 1]
```diff
@@ -44,7 +46,15 @@ func configureDjango(sourceDir string, config *ScannerConfig) (*SourceInfo, erro
 	    vars["venv"] = true
 	}
 
-	s.Files = templatesExecute("templates/django", vars)
+    wsgis, err := zglob.Glob(`./**/wsgi.py`)
+
+    if err == nil || len(wsgis) == 1 {
+        wsgiPath := strings.Split(wsgis[0], "/")
```
[the remark this answers is not in the document]
Comment (katiayn) [reply scanner/django.go:52]: I'll also check that. Thanks for the suggestion!

File: scanner/django.go [hunk 2]
```diff
@@ -44,7 +46,15 @@ func configureDjango(sourceDir string, config *ScannerConfig) (*SourceInfo, erro
 	    vars["venv"] = true
 	}
 
-	s.Files = templatesExecute("templates/django", vars)
+    wsgis, err := zglob.Glob(`./**/wsgi.py`)
+
+    if err == nil || len(wsgis) == 1 {
```
[the remark this answers is not in the document]
Comment (dangra) [reply scanner/django.go:51]: IMHO I "safe enough" thing to do when it founds more than one wsgi is to pick the first and warn about the others. sounds good?

[the remark this answers is not in the document]
Comment (katiayn) [reply scanner/django.go:51]: Yes, it sounds good! 👍🏽 I'll update the PR and let you know 🙂 Thanks for all the help!

File: scanner/django.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (katiayn) [reply scanner/django.go:52]: @dangra I was checking zglob.Glob and I believe it always returns the path with forward slashes (`/`). Maybe I'm wrong here but if not, my pattern won't work on Windows:
```
wsgis, err := zglob.Glob(`./**/wsgi.py`)
```

I can still use `path.Split` anyways:
```
dirPath, _ := path.Split(wsgis[0])
dirName := path.Base(dir)
```

Does it make sense?

[the remark this answers is not in the document]
Comment (dangra) [reply scanner/django.go:52]: I would prefer to stay on the safe lane and use path.Split whenever possible

[the remark this answers is not in the document]
Comment (dangra) [reply scanner/django.go:52]: `path.Base()` could be all we need here, right? 

oh wait, why are we splitting the path at all?

I think you want wsgiName to be the complete relative path to the .wsgi file so gunicorn can find inside the container under /code folder.

[the remark this answers is not in the document]
Comment (katiayn) [reply scanner/django.go:52]: No, I need only the folder's name (which is usually the project name) where the wsgi.py file is placed ([location of a module containing a WSGI application object named application](https://docs.djangoproject.com/en/4.2/howto/deployment/wsgi/gunicorn/#running-django-in-gunicorn-as-a-generic-wsgi-application))

The WSGI web server uses only one `application` ([for example](https://github.com/fly-apps/hello-django/blob/main/hello_django/wsgi.py#L16)):
```
CMD ["gunicorn", "--bind", ":8000", "--workers", "2", "{{ .wsgiName }}.wsgi"]
```

For example, for `/code/hello_django/wsgi.py`, the command will be like [this](https://github.com/fly-apps/hello-django/blob/main/Dockerfile#L26):
```
CMD ["gunicorn", "--bind", ":8000", "--workers", "2", "hello_django.wsgi"]
```

[the remark this answers is not in the document]
Comment (dangra) [reply scanner/django.go:52]: Okay!

[the remark this answers is not in the document]
Comment (katiayn) [reply scanner/django.go]: @dangra fixed! Thanks for the help 🙂

**Review by dangra — APPROVED** [review]
well done!

**Pull request merged**